Source/Core/DiscIO/VolumeVerifier.cpp

```diff
@@ -736,55 +757,90 @@ void VolumeVerifier::Process()
     {
       if (m_hashes_to_calculate.crc32)
       {
-        // It would be nice to use crc32_z here instead of crc32, but it isn't available on Android
-        m_crc32_context =
-            crc32(m_crc32_context, data.data(), static_cast<unsigned int>(bytes_to_read));
+        m_crc32_future = std::async(std::launch::async, [this] {
+          // Would be nice to use crc32_z here instead of crc32, but it isn't available on Android
+          m_crc32_context =
+              crc32(m_crc32_context, m_data.data(), static_cast<unsigned int>(m_data.size()));
+        });
       }
 
       if (m_hashes_to_calculate.md5)
-        mbedtls_md5_update_ret(&m_md5_context, data.data(), bytes_to_read);
+      {
+        m_md5_future = std::async(std::launch::async, [this] {
+          mbedtls_md5_update_ret(&m_md5_context, m_data.data(), m_data.size());
+        });
+      }
 
       if (m_hashes_to_calculate.sha1)
-        mbedtls_sha1_update_ret(&m_sha1_context, data.data(), bytes_to_read);
+      {
+        m_sha1_future = std::async(std::launch::async, [this] {
+          mbedtls_sha1_update_ret(&m_sha1_context, m_data.data(), m_data.size());
+        });
+      }
     }
   }
 
   if (content_read)
   {
-    if (!read_succeeded || !m_volume.CheckContentIntegrity(content, data, m_ticket))
-    {
-      AddProblem(
-          Severity::High,
-          StringFromFormat(Common::GetStringT("Content %08x is corrupt.").c_str(), content.id));
-    }
+    m_content_future = std::async(std::launch::async, [this, read_succeeded, content] {
+      if (!read_succeeded || !m_volume.CheckContentIntegrity(content, m_data, m_ticket))
```

Comment:
1) `read_succeeded` can possibly be moved outside the future so it only executes conditionally.
2) Other `m_volume` methods are protected by `m_volume_mutex`, I assume it's not needed here?

Reply:
1. The reason I didn't do that was because then I'd need to duplicate the call to `AddProblem`. A read failing is a rare event, and a WAD file usually doesn't have more than 10 contents or so anyway, so this won't have any noticeable performance impact.
2. It's only needed if the volume is reading data from the disk, which is not the case here since we pass in a vector with the data to verify. It's the same if you look at the two different calls to `CheckBlockIntegrity`.

Reply:
LGTM then